oci/spec_opts.go

```diff
@@ -365,6 +366,14 @@ func WithImageConfigArgs(image Image, args []string) SpecOpts {
 			return fmt.Errorf("unknown image config media type %s", ic.MediaType)
 		}
 
+		//FreeBSD has support for running Linux binaries via syscall redirection, while that would be enough
+		//for some binaries others depend on emulated linux file systems (procfs & sysfs).
+		//So when we are running FreeBSD and our image OS is Linux we modify our mounts to mount the emulated
+		//filesystems.
+		//We do it here since here we have access to both the OCI spec & the image OS.
+		if runtime.GOOS == "freebsd" && ociimage.OS == "linux" {
```

Comment:
The other maintainers might feel differently about this so I'm not asking you to make a change yet, but the way I probably would have done this is to define a function like `appendOSMounts` in each of `spec_opts_freebsd.go`/`spec_opts_linux.go`/`spec_opts_windows.go` with whatever content is appropriate there.  The Linux one would likely be empty, but I imagine there could be some use for this on Windows with the LCOW/WCOW work.

In addition to providing a possibly-reusable abstraction for FreeBSD and Windows to achieve similar goals, an advantage is keeping FreeBSD-specific filesystems like "linprocfs" and "linsysfs" out of the common file that is compiled for all operating systems.

The other advantage of this is that, when compiled for FreeBSD, we could also add a `WithLinuxEmulationMounts` functional option that clients of containerd can use without having to use `WithImageConfig`/`WithImageConfigArgs` or even having an image with a declared OS of "linux".

Reply:
Yes, that seems like a better approach. I'll wait for feedback from the other maintainers.

Reply:
Does anybody else want to way in on this? I see no issues in rewriting as to Samuel's suggestion.

Reply:
Samuel's suggestion seems reasonable to me.

Reply:
Since no one else has said anything: I'd like to see the change that I suggested. :smile: